File: static/codes/cn/2022-01-27-content-based-image-retrieval/sift-key-points.py
# %%
from sys import flags
import numpy as np
import cv2 as cv

from copy import deepcopy

# %%
img = cv.imread('../../../static/images/cn/2022-01-27-content-based-image-retrieval/home.jpg')
gray_img = cv.cvtColor(img, cv.COLOR_RGB2GRAY)

# %%
sift = cv.SIFT_create()

# %%
key_points = sift.detect(gray_img, None)

# %%
ret_img = deepcopy(img)
ret_img = cv.drawKeypoints(gray_img, key_points, ret_img, flags=cv.DRAW_MATCHES_FLAGS_DRAW_RICH_KEYPOINTS)

# %%
cv.imwrite('../../../static/images/cn/2022-01-27-content-based-image-retrieval/home-sift-key-points.jpg', ret_img)

# %%

File: static/codes/cn/2022-01-27-content-based-image-retrieval/sift-key-points.R
library(EBImage)

home_img <- readImage('../../../static/images/cn/2022-01-27-content-based-image-retrieval/home.jpg')
home_sift_key_points_img <- readImage('../../../static/images/cn/2022-01-27-content-based-image-retrieval/home-sift-key-points.jpg')

home_images <- combine(
    home_img,
    home_sift_key_points_img
)

png(filename = '../../../static/images/cn/2022-01-27-content-based-image-retrieval/home-raw-and-sift-key-points.png', width = 14, height = 5, units = 'cm', res = 150)
old_par <- par()
par(mar = c(1, 1, 1, 1))
display(home_images, method = 'raster', all = T, spacing = 30)
par(old_par)
dev.off()